src/sql/workbench/parts/notebook/cellViews/stdin.component.ts

```diff
@@ -0,0 +1,120 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./stdin';
+import {
+	Component, Input, Inject, ChangeDetectorRef, forwardRef,
+	ViewChild, ElementRef, AfterViewInit
+} from '@angular/core';
+import { nb } from 'azdata';
+import { localize } from 'vs/nls';
+
+import { InputBox } from 'sql/base/browser/ui/inputBox/inputBox';
+import { attachInputBoxStyler } from 'sql/platform/theme/common/styler';
+
+import { IInputOptions } from 'vs/base/browser/ui/inputbox/inputBox';
+import { IWorkbenchThemeService } from 'vs/workbench/services/themes/common/workbenchThemeService';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { inputBackground, inputBorder } from 'vs/platform/theme/common/colorRegistry';
+import { StandardKeyboardEvent, IKeyboardEvent } from 'vs/base/browser/keyboardEvent';
+import { KeyCode } from 'vs/base/common/keyCodes';
+import { AngularDisposable } from 'sql/base/node/lifecycle';
+import * as DOM from 'vs/base/browser/dom';
+import { Deferred } from 'sql/base/common/promise';
+import { ICellModel, CellExecutionState } from 'sql/workbench/parts/notebook/models/modelInterfaces';
+
+export const STDIN_SELECTOR: string = 'stdin-component';
+@Component({
+	selector: STDIN_SELECTOR,
+	template: `
+		<div class="prompt">{{prompt}}</div>
+		<div #input class="input"></div>
+	`
+})
+export class StdInComponent extends AngularDisposable implements AfterViewInit {
+	private _input: InputBox;
+	@ViewChild('input', { read: ElementRef }) private _inputContainer: ElementRef;
+
+	@Input() stdIn: nb.IStdinMessage;
+	@Input() onSendInput: Deferred<string>;
+	@Input() cellModel: ICellModel;
+
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private changeRef: ChangeDetectorRef,
+		@Inject(IWorkbenchThemeService) private themeService: IWorkbenchThemeService,
+		@Inject(IContextViewService) private contextViewService: IContextViewService,
+		@Inject(forwardRef(() => ElementRef)) private el: ElementRef
+	) {
+		super();
+	}
+
+	ngOnInit(): void {
+
+	}
+
+	ngAfterViewInit(): void {
+		let inputOptions: IInputOptions = {
+			placeholder: '',
+			ariaLabel: this.prompt
+		};
+		this._input = new InputBox(this._inputContainer.nativeElement, this.contextViewService, inputOptions);
+		if (this.password) {
+			this._input.inputElement.type = 'password';
+		}
+		this._register(this._input);
+		this._register(attachInputBoxStyler(this._input, this.themeService, {
+			inputValidationInfoBackground: inputBackground,
+			inputValidationInfoBorder: inputBorder,
+		}));
+		if (this.cellModel) {
+			this._register(this.cellModel.onExecutionStateChange((status) => this.handleExecutionChange(status)));
```

Comment:
Just trying to follow this. When we re-run a cell that prompts for StdIn what happens? The component gets recreated and whatever was in the input box goes away, right?

Reply:
Well we don't support re-run, we support cancel and then run right now, but if we did support re-run:
1. Existing run gets stopped. State changes from running to Stopped, StdIn is removed.
2. Run cell happens, stdin request received, appears with new (empty) contents